app/src/main/java/com/globalaccelerex/mpossdksample/viewModel/ItemViewModel.kt

```diff
@@ -0,0 +1,64 @@
+package com.globalaccelerex.mpossdksample.viewModel
+
+import android.content.Context
+import android.util.Log
+import androidx.lifecycle.*
+import com.globalaccelerex.mpossdksample.data.DataSource.loadCatalogListItems
+import com.globalaccelerex.mpossdksample.model.CatalogItem
+import com.globalaccelerex.mpossdksample.model.CheckoutItem
+
+
+class ItemViewModel(context: Context) : ViewModel() {
+    // List of all catalog items
+    private val _itemsList = MutableLiveData<MutableList<CatalogItem>>()
+    val itemsList: LiveData<MutableList<CatalogItem>>  get() = _itemsList
+
+    // List of all CheckOut Items
+    val selectedItemList = Transformations.map(itemsList) { catalogItemList ->
+        // Filter out items not added to cart
+        catalogItemList.filter {
+            it.isAdded
+        }.map {
+            // Map the added Catalog Items to Checkout Items
+            CheckoutItem(it.stringResourceTitle, it.stringResourcePrice)
+        }
+    }
+
+    val totalPrice = Transformations.map(selectedItemList) { checkOutItemList ->
+        checkOutItemList.sumOf { (context.resources.getString(it.stringResourcePrice)).toBigDecimal() }
```

Comment:
Can we have the price as part of the item class instead of a string resource? You can have it as price: BigDecimal